### PR TITLE
Upgrade scaling native mode, support auto init and cleanup.

### PR DESCRIPTION
**Pull request by azexcy**
Changes proposed in this pull request:
- Upgrade scaling native mode, support auto init and cleanup.
- Update scaling IT action yaml.

### CODE REVIEW

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/it-env.properties [hunk 1]
```diff
@@ -14,11 +14,22 @@
...
-# it.cluster.env.type=DOCKER,NATIVE
-it.cluster.env.type=DOCKER
-# it.env.mysql.version=mysql:5.7,mysql:8.0
-it.env.mysql.version=
-# it.env.postgresql.version=postgres:10-alpine,postgres:11-alpine,postgres:12-alpine,postgres:13-alpine,postgres:14-alpine
-it.env.postgresql.version=
-# it.env.opengauss.version=enmotech/opengauss:3.0.0,enmotech/opengauss:2.1.0
-it.env.opengauss.version=
+# it.type=DOCKER,NATIVE
+scaling.it.type=
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/it-env.properties:18]: `scaling.it.type` could be `scaling.it.env.type`, there might be other type later.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/env/enums/ScalingITTypeEnum.java [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 package org.apache.shardingsphere.integration.data.pipeline.env.enums;
 
-public enum ITEnvTypeEnum {
+public enum ScalingITTypeEnum {
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/env/enums/ScalingITTypeEnum.java:21]: `ScalingITTypeEnum` could be `ScalingITEnvTypeEnum`, keep the same as `scaling.it.env.type`

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/env/IntegrationTestEnvironment.java [hunk 3]
```diff
@@ -37,7 +44,7 @@ public final class IntegrationTestEnvironment {
     
     private final Properties props;
     
-    private final ITEnvTypeEnum itEnvType;
+    private final ScalingITTypeEnum itType;
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/env/IntegrationTestEnvironment.java:47]: `itType` could be recoverd to `itEnvType`, keep it the same as `scaling.it.env.type`

File: .github/workflows/it-scaling.yml [hunk 4]
```diff
@@ -87,9 +87,9 @@ jobs:
       - name: Build Scaling IT image
         run: ./mvnw -B clean install -am -pl shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling -Pit.env.docker -DskipTests
       - name: Run Scaling MySQL Integration Test
-        run: ./mvnw -nsu -B install -f shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/pom.xml -Dit.cluster.env.type=DOCKER -Dit.env.mysql.version=${{ env.mysql_version }}
+        run: ./mvnw -nsu -B install -f shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/pom.xml -Dscaling.it.type=docker -Dscaling.it.docker.mysql.version=${{ env.mysql_version }}
```
Comment (sandynz) [inline .github/workflows/it-scaling.yml:90]: `scaling.it.type` could `scaling.it.env.type`, keep it the same as `it-env.properties` configuration

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/initdb.sql [hunk 5]
```diff
@@ -18,8 +18,8 @@
...
-CREATE DATABASE IF NOT EXISTS ds_0;
-CREATE DATABASE IF NOT EXISTS ds_1;
-CREATE DATABASE IF NOT EXISTS ds_2;
-CREATE DATABASE IF NOT EXISTS ds_3;
-CREATE DATABASE IF NOT EXISTS ds_4;
+CREATE DATABASE IF NOT EXISTS scaling_it_0;
+CREATE DATABASE IF NOT EXISTS scaling_it_1;
+CREATE DATABASE IF NOT EXISTS scaling_it_2;
+CREATE DATABASE IF NOT EXISTS scaling_it_3;
+CREATE DATABASE IF NOT EXISTS scaling_it_4;
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/initdb.sql:25]: Do we need to drop existing database?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/database/MySQLContainer.java [hunk 6]
```diff
@@ -45,15 +45,15 @@ protected void configure() {
...
             addFixedExposedPort(3306, 3306);
         }
         setWaitStrategy(new JDBCConnectionWaitStrategy(() -> DriverManager.getConnection(DataSourceEnvironment.getURL(DATABASE_TYPE, "localhost", getFirstMappedPort()), "root", "root")));
     }
     
     @Override
-    public String getJdbcUrl(final String host, final int port, final String databaseName) {
-        String jdbcUrl = DataSourceEnvironment.getURL(DATABASE_TYPE, host, port, databaseName);
+    public String getJdbcUrl(final String databaseName) {
+        String jdbcUrl = DataSourceEnvironment.getURL(DATABASE_TYPE, this.getHost(), getFirstMappedPort(), databaseName);
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/framework/container/database/MySQLContainer.java:56]: `this.getHost()` could be `getHost()`

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/env/IntegrationTestEnvironment.java [hunk 7]
```diff
@@ -66,6 +73,106 @@ private Properties loadProperties() {
...
+     * @return default port
+     */
+    public int getActualDataSourceDefaultPort(final DatabaseType databaseType) {
+        switch (databaseType.getType()) {
+            case "MySQL":
+                return Integer.parseInt(props.getOrDefault("scaling.it.native.mysql.port", 3306).toString());
+            case "PostgreSQL":
+                return Integer.parseInt(props.getOrDefault("scaling.it.native.postgresql.port", 3306).toString());
+            case "openGauss":
+                return Integer.parseInt(props.getOrDefault("scaling.it.native.opengauss.port", 3306).toString());
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/env/IntegrationTestEnvironment.java:123]: Is `3306` for PostgreSQL and openGauss?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/general/MySQLGeneralScalingIT.java [hunk 8]
```diff
@@ -91,5 +98,6 @@ public void assertManualScalingSuccess() throws InterruptedException {
         assertPreviewTableSuccess("t_order_item", Arrays.asList("ds_2", "ds_3", "ds_4"));
         restoreScalingSourceWriting(jobId);
         assertRestoreScalingSourceWriting();
+        getJdbcTemplate().execute(String.format("DROP SCALING %s", jobId));
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/general/MySQLGeneralScalingIT.java:101]: If we need to drop job when it's completed, then it need to `DROP SCALING` even job fail.

Could we keep the job after it's completed, and clean it on startup?

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/base/BaseExtraSQLITCase.java [hunk 9]
```diff
@@ -18,23 +18,58 @@
...
+        }
+        try (Connection connection = ENV.getActualDataSourceConnection(getDatabaseType())) {
+            for (String each : Arrays.asList(DS_0, DS_1, DS_2, DS_3, DS_4)) {
+                try {
+                    connection.createStatement().execute(String.format("CREATE DATABASE %s", each));
+                } catch (final SQLException ex) {
+                    log.error("Error occurred when create database. error msg={}", ex.getMessage());
+                }
+            }
+        }
```
Comment (sandynz) [inline shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/base/BaseExtraSQLITCase.java:72]: Looks it's duplicated with `initdb.sql`

**Review by sandynz — CHANGES_REQUESTED** [review]
Thanks. There's some snippets could be double checked.

File: shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/initdb.sql (hunk 5 above)
Comment (azexcy) [reply shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/resources/env/mysql/initdb.sql:25]: the initdb.sql only use for docker mode, native mode not read the file

**Review by sandynz — APPROVED** [review]

**Pull request merged**